lib/api/v1/report-controller.js

```diff
@@ -66,6 +66,60 @@ module.exports = function(app, user) {
     });
   });
 
+  // Cancel batch
+  app.put('/api/v1/report/batch/cancel', user.can('edit reports'), function(req, res) {
+    Report.cancelBatch(req.body.id, function(err) {
+      if (err) res.send(err.status, err.message);
+      else res.send(200);
+    });
+  });
+
+  // Grab new batch for given user id
+  app.get('/api/v1/report/batch/:_id', user.can('edit reports'), function(req, res) {
+    Report.checkoutBatch(req.params._id, function(err, reports) {
+      if (err) res.send(err.status, err.message);
+      else res.send(200, {results: reports, total: reports.length});
+    });
+  });
+
+  // Mark selected reports as read
+  app.post('/api/v1/report/_read', user.can('edit reports'), function(req, res) {
+    if (!req.body.ids || !req.body.ids.length) return res.send(200);
+
+    Report.find({_id: {$in: req.body.ids}}, function(err, reports) {
+      if (err) return res.send(err.status, err.message);
+      if (reports.length === 0) return res.send(200);
+      var remaining = reports.length;
+      reports.forEach(function(report) {
+        // Mark each report as read only to catch it in model
+        report.markAsRead();       
+        report.save(function(err) {
+          if (err) return res.send(err.status, err.message);
+          if (--remaining === 0) return res.send(200);
+        });
+      });
+    });
+  });
+
+  // Flag selected reports
+  app.post('/api/v1/report/_flag', user.can('edit reports'), function(req, res) {
+    if (!req.body.ids || !req.body.ids.length) return res.send(200);
+
+    Report.find({_id: {$in: req.body.ids}}, function(err, reports) {
+      if (err) return res.send(err.status, err.message);
+      if (reports.length === 0) return res.send(200);
+      var remaining = reports.length;
+      reports.forEach(function(report) {
+        // Mark each report as flagged to catch it in model
+        report.flag();
+        report.save(function(err) {
+          if (err) return res.send(err.status, err.message);
+          if (--remaining === 0) return res.send(200);
+        });
+      });
+    });
+  });
```

Comment:
Seems like it should be possible to do this in one db query? Or will that mess with the event stuff?